digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java

```diff
@@ -1328,6 +1465,14 @@ public void insertMonitors(long attemptId, List<SessionMonitor> monitors)
                 " values (:projectId, :workflowName, :sessionTime)")
         void upsertAndLockSession(@Bind("projectId") int projectId,
                 @Bind("workflowName") String workflowName, @Bind("sessionTime") long sessionTime);
+
+        @SqlQuery("select id from sessions s" +
+                " where id = (" +
+                    "select session_id from session_attempts" +
+                    " where id = :attemptId" +
+                ")" +
+                " for update")
```

Comment:
`upsertAndLockSession()` uses a different condition to do the same thing? If so, we'd better use the same condition in both methods for better code maintenance.